app/models/person.rb

```diff
@@ -52,4 +66,15 @@ def links
   def details
     @content_item.content_item_data["details"]
   end
+
+  def ordered_previous_appointments
+    previous_roles = []
+    links["ordered_previous_appointments"].each do |previous_appointment|
```

Comment:
You could use a `.map` here to avoid needing the `.each` and then `.push`.

For example:

```ruby
links["ordered_previous_appointments"].each do |previous_appointment|
  role = previous_appointment["links"]["role"].first
  role["start_year"] = Time.parse(previous_appointment["details"]["started_on"]).strftime("%Y")
  role["end_year"] = Time.parse(previous_appointment["details"]["ended_on"]).strftime("%Y")
  role
end
```